exporter/signalfxexporter/translation/translator.go

```diff
@@ -87,19 +188,66 @@ func createDimensionsMap(rules []Rule) map[string]string {
 	return nil
 }
 
-func (mp *MetricTranslator) TranslateDataPoints(sfxDataPoints []*sfxpb.DataPoint) {
+func (mp *MetricTranslator) TranslateDataPoints(logger *zap.Logger, sfxDataPoints []*sfxpb.DataPoint) []*sfxpb.DataPoint {
+	processedDataPoints := sfxDataPoints
+
 	for _, tr := range mp.rules {
-		for _, dp := range sfxDataPoints {
+		newDataPoints := []*sfxpb.DataPoint{}
+
+		for _, dp := range processedDataPoints {
 			switch tr.Action {
```

Comment:
Since Action is a string this switch may also be slow. Converting Action to a numeric equivalent at config reading time can help improve performance. Again, not suggesting to do it, we can do it if it shows to be significant during benchmarking later.